lib/pagerduty/configuration.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'common/client/configuration/rest'
+require 'common/client/middleware/response/json_parser'
+require 'common/client/middleware/response/raise_error'
+
+module PagerDuty
+  class Configuration < Common::Client::Configuration::REST
+    def connection
+      Faraday.new(base_path, headers: base_request_headers, request: request_options) do |conn|
+        conn.use :breakers
+        conn.request :json
+
+        conn.response :raise_error, error_prefix: service_name
+        conn.response :json_parser
+
+        conn.adapter Faraday.default_adapter
+      end
+    end
+
+    def base_path
+      Settings.maintenance.pagerduty_api_url
+    end
+
+    def service_name
+      'PD'
```

Comment:
Small things, but I'd prefer 'PagerDuty' here as the number of acronyms is staggering :)